async_upnp_client/server.py

```diff
@@ -329,10 +329,10 @@ def _on_data(
                 self._send_responses_device_udn(remote_addr, device)
         elif matched_devices := self._matched_devices_by_type(search_target):
             for device in matched_devices:
-                self._send_responses_device_type(remote_addr, device)
+                self._send_responses_device_type(remote_addr, device, headers["st"])
         elif matched_services := self._matched_services_by_type(search_target):
             for service in matched_services:
-                self._send_responses_service(remote_addr, service)
+                self._send_responses_service(remote_addr, service, headers["st"])
```

Comment:
Use the variable `search_target` here. It is already lowered, potentially fixing a case-mismatch in `_match_type_versions()`.